presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDirectSelectiveStreamReader.java

```diff
@@ -129,27 +132,59 @@ public int read(int offset, int[] positions, int positionCount)
             skip(offset - readOffset);
         }
 
-        prepareForNextRead(positionCount, positions);
+        int dataLength = prepareForNextRead(positionCount, positions);
 
         int streamPosition;
 
         if (lengthStream == null) {
             streamPosition = readAllNulls(positions, positionCount);
         }
         else if (filter == null) {
-            streamPosition = readNoFilter(positions, positionCount);
+            streamPosition = readNoFilter(positions, positionCount, dataLength);
         }
         else {
-            streamPosition = readWithFilter(positions, positionCount);
+            streamPosition = readWithFilter(positions, positionCount, dataLength);
         }
 
         readOffset = offset + streamPosition;
         return outputPositionCount;
     }
 
-    private int readNoFilter(int[] positions, int positionCount)
+    private int readNoFilter(int[] positions, int positionCount, int dataLength)
             throws IOException
     {
+        // filter == null implies outputRequired == true
+
+        int totalPositionCount = positions[positionCount - 1] + 1;
```

Comment:
No. The positionCount comes from the batchSize in `OrcSelectiveRecordReader` and it starts with 1 and onwards. After each read, if output positionCount is 0, it will cause early termination of the StreamReader read loop in `OrcSelectiveRecordReader.getNextPage()`